### PR TITLE
Resize BeatmapCarousel and confine its PositionalInput

### LINKED ISSUE #5452: Beatmap carousel is clickable through the footer
This is purposely regressed in https://github.com/ppy/osu/pull/5440. 

It should not block positional input. Instead, the carousel should be contained within the area above the footer, and be unmasked by the footer to provide visibility.

### PR DESCRIPTION
**Pull request by nyquillerium**
* Fixes https://github.com/ppy/osu/issues/5452
* [x] Depends on https://github.com/ppy/osu-framework/pull/2694

Also makes `FilterControl` not block input (similar to the footer).

### CODE REVIEW

File: osu.Game/Screens/Select/BeatmapCarousel.cs [hunk 1]
```diff
@@ -637,12 +643,13 @@ private static float offsetX(float dist, float halfHeight)
...
         /// </summary>
         /// <param name="p">The item to be updated.</param>
-        /// <param name="halfHeight">Half the draw height of the carousel container.</param>
+        /// <param name="halfHeight">Half the draw height of the carousel container's parent.</param>
         private void updateItem(DrawableCarouselItem p, float halfHeight)
         {
             var height = p.IsPresent ? p.DrawHeight : 0;
 
-            float itemDrawY = p.Position.Y - Current + height / 2;
+            // The actual Y position of the item needs to be offset by any potential padding set by the container's parent.
```
Comment (peppy) [inline osu.Game/Screens/Select/BeatmapCarousel.cs]: it's only offset by one of two paddings, so the comment is ambiguous.

File: osu.Game/Screens/Select/Footer.cs [hunk 2]
```diff
@@ -62,10 +62,6 @@ private void showOverlay(OverlayContainer overlay)
 
         public Footer()
         {
-            RelativeSizeAxes = Axes.X;
```
Comment (peppy) [inline osu.Game/Screens/Select/Footer.cs]: why was this moved?

File: osu.Game/Screens/Select/SongSelect.cs [hunk 3]
```diff
@@ -43,6 +43,8 @@ public abstract class SongSelect : OsuScreen, IKeyBindingHandler<GlobalAction>
     {
         private static readonly Vector2 wedged_container_size = new Vector2(0.5f, 245);
 
+        public const float FILTER_CONTROL_HEIGHT = 100;
+        public const float FOOTER_HEIGHT = 50;
```
Comment (peppy) [inline osu.Game/Screens/Select/SongSelect.cs]: duplicate

**Review by peppy — CHANGES_REQUESTED** [review]

**Comment by peppy** [comment]
I would have probably added two private const to BeatmapCarousel so you aren't constantly referencing external constants

const float bleed_top = FilterControl.HEIGHT;
const float bleed_bottom = Footer.HEIGHT;

File: osu.Game/Screens/Select/Footer.cs (hunk 2 above)
Comment (nyquillerium) [reply osu.Game/Screens/Select/Footer.cs]: I moved this because I was under the impression we should be avoiding having components resize themselves, and Footer had its constant in the class while `FilterControl` had it in `SongSelect`. Judging from your comments, i'm guessing you want it the other way around.

Comment (peppy) [reply osu.Game/Screens/Select/Footer.cs]: Components regularly size themselves. it's resizing that's an issue.

**Comment by nyquillerium** [comment]
I've applied the suggestions, but I'm going to try and find out if I can clean up the fade/positioning logic a bit more.

**Comment by nyquillerium** [comment]
I've combined the new bleed calculations into properties to make them read better, and in the process discovered/fixed some issues. Not sure if this is much better than before though.

**Review by peppy — APPROVED** [review]
